File: app_manager/appmanager/src/main/java/app/editors/manager/managers/retrofit/WebDavInterceptor.java
package app.editors.manager.managers.retrofit;

import androidx.annotation.NonNull;

import java.io.IOException;

import javax.inject.Inject;

import app.editors.manager.app.App;
import app.editors.manager.app.WebDavApi;
import app.editors.manager.managers.tools.PreferenceTool;
import okhttp3.Credentials;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class WebDavInterceptor implements Interceptor {

    @Inject
    PreferenceTool mPreferenceTool;

    public WebDavInterceptor() {
        App.getApp().getAppComponent().inject(this);
    }

    @NonNull
    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {

        String auth = chain.request().header(WebDavApi.HEADER_AUTHORIZATION);

        Request request;

        if (auth == null) {
            request = chain.request().newBuilder()
                    .addHeader(WebDavApi.HEADER_AUTHORIZATION, Credentials.basic(mPreferenceTool.getLogin(), mPreferenceTool.getPassword()))
                    .build();
        } else {
            request = chain.request();
        }

        return chain.proceed(request);
    }
}

File: app_manager/core/src/main/java/app/documents/core/settings/WebDavInterceptor.kt
package app.documents.core.settings

import app.documents.core.webdav.WebDavApi
import okhttp3.Credentials
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

class WebDavInterceptor(val login: String?, val password: String?) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        if (login == null && password == null) {
            return chain.proceed(chain.request())
        } else {
            val auth = chain.request().header(WebDavApi.HEADER_AUTHORIZATION)
            val request: Request = if (auth == null) {
                chain.request().newBuilder()
                    .addHeader(
                        WebDavApi.HEADER_AUTHORIZATION,
                        Credentials.basic(login ?: "", password ?: "")
                    )
                    .build()
            } else {
                chain.request()
            }

            return chain.proceed(request)
        }
    }

}